src/Psalm/Internal/Type/Comparator/ScalarTypeComparator.php

```diff
@@ -291,6 +289,20 @@ public static function isContainedBy(
             return true;
         }
 
+        if ($container_type_part instanceof TArrayKey
+            && $input_type_part instanceof TNumeric
+        ) {
+            return true;
+        }
+
+        if ($container_type_part instanceof TArrayKey
+            && ($input_type_part instanceof TInt
+                || $input_type_part instanceof TString
+                || $input_type_part instanceof TTemplateKeyOf)
```

Comment:
This specific line is redundant, it would have been handled line 270

Reply:
There is now also `TValueOfArray::isViableTemplateType()` and used there.